sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -854,11 +857,23 @@ object SparkSession extends Logging {
      *
      * @since 2.0.0
      */
+    @deprecated("use conf", "3.3.1")
     def config(key: String, value: Boolean): Builder = synchronized {
       options += key -> value.toString
       this
     }
 
+    /**
+     * Sets a config option. Options set using this method are automatically propagated to
+     * both `SparkConf` and SparkSession's own configuration.
+     *
+     * @since 3.3.1
+     */
+    def conf(key: String, value: Any): Builder = synchronized {
```

Comment:
Hm, does this work from Java? I'm concerned that this method becomes harder to use outside Scala

Reply:
@srowen 
yes. it is works!

this method is simple code.
but, I afraid another type of args into this (without boolean, long, string)

Reply:
I don't quite understand. The type in the bytecode is Object right? So I can't pass a long in Java. Scala figures it out with boxing

Reply:
@srowen 
I added test code. (java) https://github.com/apache/spark/pull/37478/commits/14b6fe1349ab477546a30e99449e7a8359bd3d3b

we will check github workflows. (build - test)